### PR TITLE
new: dev: Add a Enviornment custom nml class extension

### PR DESCRIPTION
**Pull request by jcrono**
As discussed on HPENetworking/topology#67, implementing the environment info as a NML extension class.

This would allow HPENetworking/topology#55 to advance.

### CODE REVIEW

File: lib/pynml/spec.py [hunk 1]
```diff
@@ -590,6 +590,20 @@
...
+            'parent': None,
+            'brief': 'Describes attributes inherent to the environment',
+            'doc': (
+                'Attributes to be attached to the environment the topology is'
+                'in.'
+            ),
+            'abstract': False,
+            'attributes': [
+            ],
+            'relations': [
```
Comment (saenzpa) [inline lib/pynml/spec.py]: An environment should have a relation to a topology. Based off "2.2.7 hasNode" please create a new relation hasTopology that bounds 1 environment to N topologies.

Comment (jcrono) [reply lib/pynml/spec.py]: Done, but instead Topology has a relationship hasEnvironment

File: lib/pynml/nml.py [hunk 2]
```diff
@@ -2401,6 +2448,18 @@ def get_has_link(self):
         return copy(self._has_link_links)
 
 
+class Environment(NMLObject):
+    """
+    Describes attributes inherent to the environment.
+
+    Attributes to be attached to the environment the topology isin..
```
Comment (ocelotl) [inline lib/pynml/nml.py]: Typo here

File: lib/pynml/nml.py [hunk 3]
```diff
@@ -2401,6 +2448,18 @@ def get_has_link(self):
...
+    """
+
+    def __init__(
```
Comment (ocelotl) [inline lib/pynml/nml.py]: Unnecessary indentation

Comment (ocelotl) [inline lib/pynml/nml.py]: In fact, a method that only calls ``super`` with the exact same arguments than its ancestor is unnecessary

File: lib/pynml/spec.py [hunk 4]
```diff
@@ -452,6 +452,12 @@
                     'cardinality': '+',
                     'doc': 'FIXME: Document hasService relation'
                 },
+                {
+                    'name': 'hasEnvironment',
+                    'with': ['Environment'],
+                    'cardinality': '+',
+                    'doc': 'FIXME: Document hasEnvironment relation'
```
Comment (ocelotl) [inline lib/pynml/spec.py]: Fix this FIXME

File: lib/pynml/spec.py [hunk 5]
```diff
@@ -590,6 +596,20 @@
                 }
             ]
         },
+        {
+            'name': 'Environment',
+            'parent': None,
+            'brief': 'Describes attributes inherent to the environment',
+            'doc': (
+                'Attributes to be attached to the environment the topology is'
```
Comment (ocelotl) [inline lib/pynml/spec.py]: Missing space at the end of this string

**Comment by jcrono** [comment]
Done

**Comment by jcrono** [comment]
Ok so I have made ExtendendNMLManager's environment private and allow only read access through a property publicly, this is more consistent with the rest of the API

**Pull request merged**